### PR TITLE
[stable10] In case the email delivery fails the wrong status was returned.

### PR DESCRIPTION
**Pull request by phil-davis**
Backport #32218

### CODE REVIEW

**Comment by phil-davis** [comment]
This had the backport-request label but had not been backported.

**Review by PVince81 — APPROVED** [review]
:+1:

File: apps/dav/tests/unit/CalDAV/Schedule/IMipPluginTest.php [hunk 1]
```diff
@@ -29,17 +29,18 @@
 use Sabre\VObject\Component\VCalendar;
 use Sabre\VObject\ITip\Message;
 use Test\TestCase;
+use OC\Log;
 
 class IMipPluginTest extends TestCase {
-	public function testDelivery() {
+	public function testDelivery(): void {
```
Comment (phil-davis) [inline apps/dav/tests/unit/CalDAV/Schedule/IMipPluginTest.php]: PHP 5.6 lint does not like ``: void`` on this line.
https://drone.owncloud.com/owncloud/core/9844/68
```
Checked 2956 files in 31.6 seconds
Syntax error found in 1 file

------------------------------------------------------------
Parse error: ./apps/dav/tests/unit/CalDAV/Schedule/IMipPluginTest.php:35
    33| 
    34| class IMipPluginTest extends TestCase {
  > 35| 	public function testDelivery(): void {
    36| 		$mailMessage = new \OC\Mail\Message(new \Swift_Message());
    37| 		/** @var Mailer | \PHPUnit_Framework_MockObject_MockObject $mailer */
Unexpected ':', expecting ';' or '{' in ./apps/dav/tests/unit/CalDAV/Schedule/IMipPluginTest.php on line 35
Makefile:203: recipe for target 'test-php-lint' failed
```

Comment (PVince81) [reply apps/dav/tests/unit/CalDAV/Schedule/IMipPluginTest.php]: can you remove it ?

Comment (phil-davis) [reply apps/dav/tests/unit/CalDAV/Schedule/IMipPluginTest.php]: It was added by this PR. But I guess it is PHP 7 syntax, so we cannot backport exactly stufff like this.
I will look...

**Comment by phil-davis** [comment]
Pushed a 2nd commit to remove the PHP7 feature.
http://php.net/manual/en/functions.returning-values.php#functions.returning-values.type-declaration

**Pull request merged**